src/core/thread/mle.cpp

```diff
@@ -2391,6 +2558,66 @@ ThreadError Mle::HandleChildIdResponse(const Message &aMessage, const Ip6::Messa
     return error;
 }
 
+ThreadError Mle::HandleChildUpdateRequest(const Message &aMessage, const Ip6::MessageInfo &aMessageInfo)
+{
+    static const uint8_t kMaxResponseTlvs = 5;
+
+    ThreadError error = kThreadError_None;
+    SourceAddressTlv sourceAddress;
+    LeaderDataTlv leaderData;
+    NetworkDataTlv networkData;
+    ChallengeTlv challenge;
+    TlvRequestTlv tlvRequest;
+    uint8_t tlvs[kMaxResponseTlvs];
+    uint8_t numTlvs = 0;
+
+    otLogInfoMle("Received Child Update Request from parent");
+
+    // Source Address
+    SuccessOrExit(error = Tlv::GetTlv(aMessage, Tlv::kSourceAddress, sizeof(sourceAddress), sourceAddress));
+    VerifyOrExit(sourceAddress.IsValid(), error = kThreadError_Parse);
+    VerifyOrExit(mParent.mValid.mRloc16 == sourceAddress.GetRloc16(), error = kThreadError_Drop);
+
+    // Leader Data
+    if (Tlv::GetTlv(aMessage, Tlv::kLeaderData, sizeof(leaderData), leaderData) == kThreadError_None)
+    {
+        VerifyOrExit(leaderData.IsValid(), error = kThreadError_Parse);
+        SetLeaderData(leaderData.GetPartitionId(), leaderData.GetWeighting(), leaderData.GetLeaderRouterId());
+    }
+
+    // Network Data
+    if (Tlv::GetTlv(aMessage, Tlv::kNetworkData, sizeof(networkData), networkData) == kThreadError_None)
+    {
+        VerifyOrExit(networkData.IsValid(), error = kThreadError_Parse);
+        mNetworkData.SetNetworkData(leaderData.GetDataVersion(), leaderData.GetStableDataVersion(),
```

Comment:
If the network data version or stable version changes (depending on whether this device is requesting full or stable network data), the device will also need to send a MLE Data Request message to retrieve possible updates to the Active and Pending Datasets.  See `Mle::HandleChildUpdateResponse` for a similar example.